k8s/configuration/secret.yaml

```diff
@@ -0,0 +1,92 @@
+# All secret data should be base64 encoded before embedding them here by
+# using `echo "secret string" | base64 -w 0 > secret.string.b64` and then
+# copy the resulting value here.
+# Ref: https://kubernetes.io/docs/concepts/configuration/secret/
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mdb-agent-api-key
+  namespace: default
+type: Opaque
+data:
+  # This is the API Key obtained from MongoDB Cloud Manager
+  api-key: "<b64 encoded api key>"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bdb-private-key
+  namespace: default
+type: Opaque
+data:
+  private.key: "<b64 encoded private key>"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mdb-certs
+  namespace: default
+type: Opaque
+data:
+  mdb-instance.pem: "<b64 encoded, concatanated public and private keys>"
+  ca.pem: "<b64 encoded CA public key>" 
+  mdb-crl.pem: "<b64 encoded CRL data>" 
```

Comment:
I would have thought that the CRL was something associated with the CA, not just MongoDB (although the MongoDB instance may be the only one that actually uses the CRL). Maybe we could move the CRL pem to the `ca` Secret suggested earlier, along with `ca.pem`? I would also rename it just `crl.pem`.